### ISSUE TITLE
Is $\tau^2$ in Eq.(8) missed?

### ISSUE THREAD

**Issue by yuleiniu** [caiqi/Joint-Contrastive-Learning#1]
Thanks for your interesting and inspiring work and codebase! 

It seems that $\tau^2$ in Eq.(8) is missed in [moco/moco_unlimit_key_default.py](https://github.com/caiqi/Joint-Contrastive-Learning/blob/master/moco/moco_unlimit_key_default.py) since 'self.T' (i.e., $\tau$) is used only once in line 167 and 175.

I wonder whether it is correct to 
(1) change line 159 from
`[q, k_means + 0.5 * torch.bmm(k_sigma, q.unsqueeze(dim=-1)).squeeze(dim=-1)]).unsqueeze(-1)`
to 
`[q, k_means + 0.5 * torch.bmm(k_sigma, q.unsqueeze(dim=-1)).squeeze(dim=-1) / self.T]).unsqueeze(-1)`

and change line 175 from 
`isda_loss = 0.5 * torch.bmm(torch.bmm(q.reshape(N, 1, C), k_sigma), q.reshape(N, C, 1)).mean() / self.T`
to
`isda_loss = 0.5 * torch.bmm(torch.bmm(q.reshape(N, 1, C), k_sigma), q.reshape(N, C, 1)).mean() / (self.T^2)`

OR

(2) change line 152
from
`k_sigma = k_sigma * ratio`
to
`k_sigma = k_sigma * ratio / self.T`

Thank you.

**Comment by caiqi** [comment]
Thanks for your interest in our work! Regarding the code, please note each term in the logits has been scaled by \tau. In this way, the effect of the squared \tau can be absorbed into the value of \lambda. The effective \lambda used in our current implementation is therefore scaled by \fraction{1}{\tau}. In our paper presentation, we have correspondingly modified the scale of $\lambda$ by using the value of $\lambda^'$=$\lambda \times \tau$, and this modification has no effect on the final scores reported on various downstream tasks. We would soon update this code to clarify.

**Comment by yuleiniu** [comment]
I get it. Thank you.
